src/eva/vision/models/networks/from_timm.py

```diff
@@ -0,0 +1,49 @@
+"""Helper wrapper class for timm models."""
+
+from typing import Any, Dict, List
+
+import timm
+import torch
+from torch import nn
+from typing_extensions import override
+
+
+class TimmModel(nn.Module):
+    """Wrapper class for timm models."""
+
+    def __init__(
+        self,
+        model_name: str,
+        pretrained: bool = False,
+        checkpoint_path: str = "",
+        model_arguments: Dict[str, Any] | None = None,
+    ) -> None:
+        """Initializes and constructs the model.
+
+        Args:
+            model_name: Name of model to instantiate.
+            pretrained: If set to `True`, load pretrained ImageNet-1k weights.
+            checkpoint_path: Path of checkpoint to load.
+            model_arguments: The extra callable function / class arguments.
+        """
+        super().__init__()
+
+        self._model_name = model_name
+        self._pretrained = pretrained
+        self._checkpoint_path = checkpoint_path
+        self._model_arguments = model_arguments or {}
+
+        self._feature_extractor = self._load_model()
+
+    def _load_model(self) -> nn.Module:
+        """Builds, loads and returns the model."""
+        return timm.create_model(
```

Comment:
Couldn't we use use the existing `ModelFromFunction` in conjunction with `timm.create_model`?

Reply:
I though maybe it would be easier for the users, which the majority of them will use the `TimmModels` to have a dedicated wrapper, since the `ModelFromFunction` will make them undestund that the model from timm initialized with a function, in yaml we can not init a function that's why we have to use the wrapper `ModelFromFunction`

what do you think?

Reply:
Good point. I think it's fine to have a dedicated wrapper for Timm, but we should use the wrapper api from eva.core I think?

Reply:
oh yes no problem!

Reply:
done